### PR TITLE
fix exception in kg query

### PR DESCRIPTION
**Pull request by fengsh27**
### Abstract
When kg query got no results (question was not related to kg), it would throw exception, which will break the chat. To fix this issue, we need to catch this exception and leave context empty.

### CODE REVIEW

File: biochatter/llm_connect.py [hunk 1]
```diff
@@ -257,18 +260,26 @@ def _inject_context(self, text: str):
                 for agent in self.rag_agents:
                     if not agent.use_prompt:
                         continue
-                    statements = statements + [
-                        doc[0] for doc in agent.generate_responses(text)
-                    ]
+                    try:
+                        docs = agent.generate_responses(text)
+                        statements = statements + [
+                            doc[0] for doc in agent.generate_responses(text)
```
Comment (slobentanzer) [inline biochatter/llm_connect.py:266]: shouldn't this be `for doc in docs`, as below?

Comment (fengsh27) [reply biochatter/llm_connect.py:266]: Good catch.

File: biochatter/llm_connect.py [hunk 2]
```diff
@@ -257,18 +260,26 @@ def _inject_context(self, text: str):
...
+                        ]
+                    except Exception as e:
```
[the remark this answers is not in the document]
Comment (fengsh27) [reply biochatter/llm_connect.py:268]: make sense

**Review by slobentanzer — APPROVED** [review]
Thanks!

**Pull request merged**